odoo/fields.py

```diff
@@ -1536,11 +1543,15 @@ class _String(Field):
     translate = False                   # whether the field is translated
     prefetch = None
     unaccent = True
+    pattern = None                      # regex to validate an input
+    pattern_flags = 0                   # flags used for the regex validation
```

Comment:
I don't see any usage of this.
IMO, it adds some complexity for nothing.

Reply:
I don't really care about the flags, they were asked by the framework's reviewers

https://github.com/odoo/odoo/pull/74868#issuecomment-969044334
https://github.com/odoo/odoo/pull/74868#discussion_r749420134